dali/python/nvidia/dali/external_source.py

```diff
@@ -119,9 +159,11 @@ class ExternalSource():
 ------------
 
 `cycle`: bool
-    If `True`, the source iterable will be wrapped. Otherwise, StopIteration error wil be raised
-    when end of data is reached. Setting this flag to True when `source` is not an iterable is an
-    error.
+    If `True`, the source will be wrapped. Otherwise, StopIteration error wil be raised
+    when end of data is reached. This flag requires that `source` is either a collection, i.e. an
+    iteratble object where ``iter(source)`` will return a fresh iterator on each call or a
+    generator function. In the latter case, the generator function will be called again when more
+    data is requried than was yielded by the function.
```

Comment:
```suggestion
    If ``True``, the source will be wrapped. Otherwise, ``StopIteration`` will be raised
    when end of data is reached. This flag requires that ``source`` is either a collection, i.e. an
    iterable object where ``iter(source)`` will return a fresh iterator on each call, or a
    generator function. In the latter case, the generator function will be called again when more
    data is requested than was yielded by the function.
```
My suggestions:
1. I'm not sure if we should call `StopIteration` "error", since it's regular python flow-control mechanism
2. typo in `wil`
3. typo in `iteratble`
4. `requried` -> `requested`
5. I'd wrap argument names etc. in double-back-ticks. It renders better, IMHO